arbnode/espresso_caff_node_test.go

```diff
@@ -87,29 +98,36 @@ func (m *MockDelayedMessageFetcher) reset(parentChainBlockNum uint64, seqNum uin
 func TestEspressoCaffNodeShouldReadDelayedMessageFromL1(t *testing.T) {
 
 	caffNode := EspressoCaffNode{}
-	caffNode.espressoStreamer = &MockEspressoStreamer{delayedPos: 3}
+	caffNode.espressoStreamer = &MockEspressoStreamer{delayedPos: 3, currrPos: 0}
 	caffNode.delayedMessageFetcher = &MockDelayedMessageFetcher{}
 	ctx := context.Background()
-	msg1, err := caffNode.peekMessage(context.Background())
+	msg1, err := caffNode.peekMessage(ctx)
 	require.NoError(t, err)
 
 	require.Equal(t, msg1.MessageWithMeta.DelayedMessagesRead, uint64(1))
 	require.Equal(t, msg1.MessageWithMeta.Message, &arbostypes.EmptyTestIncomingMessage)
+	require.Equal(t, msg1.Pos, uint64(0))
+	caffNode.espressoStreamer.Advance()
 
 	msg2, err := caffNode.peekMessage(ctx)
 	require.NoError(t, err)
 	require.Equal(t, msg2.MessageWithMeta.DelayedMessagesRead, uint64(1))
 	require.Equal(t, msg2.MessageWithMeta.Message, &arbostypes.EmptyTestIncomingMessage)
+	require.Equal(t, msg2.Pos, uint64(1))
+	caffNode.espressoStreamer.Advance()
 
 	msg3, err := caffNode.peekMessage(ctx)
 	require.NoError(t, err)
 	require.Equal(t, msg3.MessageWithMeta.DelayedMessagesRead, uint64(1))
 	require.Equal(t, msg3.MessageWithMeta.Message, &arbostypes.EmptyTestIncomingMessage)
+	require.Equal(t, msg3.Pos, uint64(2))
+	caffNode.espressoStreamer.Advance()
 
 	msg4, err := caffNode.peekMessage(ctx)
 	require.NoError(t, err)
 	require.Equal(t, msg4.MessageWithMeta.DelayedMessagesRead, uint64(2))
-	require.Equal(t, msg4.MessageWithMeta.Message, &arbostypes.EmptyTestIncomingMessage)
+	require.Equal(t, msg4.MessageWithMeta.Message, arbostypes.InvalidL1Message)
```

Comment:
Here intetionally tested if the caff node replace the message content from hotshot with the one fetched from L1. Check the `processDelayedMessage` function and. you will know how it works.